src/app/Models/Event.php

```diff
@@ -6,103 +6,129 @@
 use Illuminate\Support\Facades\Auth;
 use LaravelEnso\Core\app\Models\User;
 use Illuminate\Database\Eloquent\Model;
-// use LaravelEnso\Calendar\app\Enums\Frequencies;
 use LaravelEnso\TrackWho\app\Traits\CreatedBy;
 use LaravelEnso\Helpers\app\Traits\DateAttributes;
 use LaravelEnso\Calendar\app\Contracts\ProvidesEvent;
+use LaravelEnso\Calendar\app\Services\Frequency\Create;
+use LaravelEnso\Calendar\app\Services\Frequency\Delete;
+use LaravelEnso\Calendar\app\Services\Frequency\Update;
+use LaravelEnso\Calendar\app\Contracts\Calendar as CalendarContract;
 
 class Event extends Model implements ProvidesEvent
 {
     use CreatedBy, DateAttributes;
 
+    protected $table = 'calendar_events';
+
     protected $fillable = [
         'title', 'body', 'calendar', 'frequence', 'location', 'lat', 'lng',
-        'starts_at', 'ends_at', 'frequence_ends_at', 'is_all_day', 'is_readonly',
+        'starts_on', 'ends_on', 'starts_time', 'ends_time', 'is_all_day',
+        'recurrence_ends_at', 'is_readonly', 'calendar_id', 'parent_id',
     ];
 
     protected $casts = ['is_all_day' => 'boolean', 'is_readonly' => 'boolean'];
 
-    protected $dates = ['starts_at', 'ends_at', 'frequence_ends_at'];
+    protected $dates = ['starts_on', 'ends_on', 'recurrence_ends_at'];
+
+    public function parent()
+    {
+        return $this->belongsTo(static::class, 'parent_id');
+    }
+
+    public function events()
+    {
+        return $this->hasMany(static::class, 'parent_id');
+    }
 
     public function attendees()
     {
         return $this->belongsToMany(User::class);
     }
 
+    public function calendar()
+    {
+        return $this->belongsTo(Calendar::class, 'calendar_id', 'id');
+    }
+
     public function attendeeList()
     {
-        return $this->attendees->pluck('id');
+        return $this->attendees->pluck('id')->toArray();
     }
 
     public function reminders()
     {
         return $this->hasMany(Reminder::class);
     }
 
-    public function setStartsAtAttribute($value)
+    public function setStartsOnAttribute($value)
     {
-        $this->fillDateAttribute('starts_at', $value, 'Y-m-d H:i:s');
+        $this->fillDateAttribute(
+            'starts_on', $value, config('enso.config.dateFormat')
+        );
     }
 
-    public function setEndsAtAttribute($value)
+    public function setEndsOnAttribute($value)
     {
-        $this->fillDateAttribute('ends_at', $value, 'Y-m-d H:i:s');
+        $this->fillDateAttribute(
+            'ends_on', $value, config('enso.config.dateFormat')
```

Comment:
`config('enso.config.dateFormat')` is not needed here, since that's what is used by default if  not given